lumicks/pylake/channel.py

```diff
@@ -131,6 +135,53 @@ def downsampled_over(self, range_list, reduce=np.mean, where='center'):
 
         return Slice(TimeSeries(d, t), self.labels)
 
+    def downsampled_to(self, frequency, reduce=np.mean, where='center', force=False):
+        """Return a copy of this slice downsampled to a specified frequency
+
+        Parameters
+        ----------
+        frequency : int
+            The desired downsampled frequency downsampled (Hz)
+        reduce : callable
+            The `numpy` function which is going to reduce multiple samples into one.
+            The default is `np.mean`, but `np.sum` could also be appropriate for some
+            cases, e.g. photon counts.
+        where : str
+            Where to put the final time point.
+            'center' time point is put at start + stop / 2
+            'left' time point is put at start
+        force : bool
+            if `True`, will allow downsampling for variable frequency slices and
+            for non-integer ratios of original and new frequencies.
+        """
+
+        def get_good_frequencies(frequency, target):
+            possible = np.arange(frequency)+1
+            choices = np.extract(np.mod(frequency, possible)==0, possible).astype(np.int)
+            ix = int(np.argwhere(choices < target)[-1])
```

Comment:
This can fail when no lower frequency exists e.g. `get_good_frequencies(1e9/850, 400)`

But actually, maybe we could have a second parameter named method rather than force.

With `force` pushing for whatever frequency the user specified. And `nearest` just getting the next achievable timestep. It would also sidestep the issue of the floating point math and get you a much closer achievable sample rate than an integer one that's suggested:
```
def get_next_timestep_size(source_timestep, target_timestep):
    remainder = target_timestep % source_timestep
    return target_timestep - remainder
```

What do you think?